### PR TITLE
Minor code cleanups

### PR DESCRIPTION
**Pull request by Broxzier**

### CODE REVIEW

File: src/openrct2-ui/WindowManager.cpp [hunk 1]
```diff
@@ -342,7 +342,6 @@ class WindowManager final : public IWindowManager
                 {
                     window_close_construction_windows();
                     _currentRideIndex = static_cast<ride_id_t>(rideIndex);
-                    w = OpenWindow(WC_RIDE_CONSTRUCTION);
```
Comment (duncanspumpkin) [inline src/openrct2-ui/WindowManager.cpp:345]: Are you sure about this change. You are now no longer opening the construction window if it isn't already opened.

File: src/openrct2-ui/interface/Widget.cpp [hunk 2]
```diff
@@ -491,8 +489,8 @@ static void WidgetGroupboxDraw(rct_drawpixelinfo* dpi, rct_window* w, rct_widget
     // Resolve the absolute ltrb
     l = w->windowPos.x + widget->left;
     t = w->windowPos.y + widget->top + 4;
-    r = w->windowPos.x + widget->right;
-    b = w->windowPos.y + widget->bottom;
+    int32_t r = w->windowPos.x + widget->right;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/Widget.cpp:492]: `const auto`

File: src/openrct2-ui/WindowManager.cpp (hunk 1 above)
Comment (Broxzier) [reply src/openrct2-ui/WindowManager.cpp:345]: Nope, that slipped in. Only the assignment to `w` is useless.

File: src/openrct2/Game.cpp [hunk 3]
```diff
@@ -530,11 +528,10 @@ void game_load_init()
     {
         viewport_init_all();
         game_create_windows();
-        mainWindow = window_get_main();
     }
     else
     {
-        mainWindow = window_get_main();
+        const auto mainWindow = window_get_main();
```
Comment (Gymnasiast) [inline src/openrct2/Game.cpp:534]: Shouldn’t this be

```suggestion
        const auto* mainWindow = window_get_main();
```
?

File: src/openrct2/ride/TrackDesign.cpp [hunk 4]
```diff
@@ -247,7 +247,6 @@ rct_string_id TrackDesign::CreateTrackDesignTrack(const Ride& ride)
         }
         trackElement.x = newCoords->x;
         trackElement.y = newCoords->y;
-        z = newCoords->z;
```
Comment (Gymnasiast) [inline src/openrct2/ride/TrackDesign.cpp:250]: `z` is used on line 290 - removing the statement changes the value of it.

File: src/openrct2/Game.cpp (hunk 3 above)
Comment (Broxzier) [reply src/openrct2/Game.cpp:534]: With and without explicit `*` seems to be a bit mixed.

```
src/openrct2-ui/WindowManager.cpp:527:        auto mainWindow = window_get_main();
src/openrct2-ui/input/InputManager.cpp:115:    auto mainWindow = window_get_main();
src/openrct2-ui/input/Shortcuts.cpp:60:        auto window = window_get_main();
src/openrct2-ui/input/Shortcuts.cpp:72:        auto window = window_get_main();
src/openrct2-ui/windows/RideConstruction.cpp:2141:            auto* mainWnd = window_get_main();
src/openrct2-ui/windows/TopToolbar.cpp:837:    const auto* mainWindow = window_get_main();
src/openrct2-ui/windows/TopToolbar.cpp:1217:                auto* mainWnd = window_get_main();
src/openrct2-ui/windows/Viewport.cpp:87:        auto* mainWindow = window_get_main();
src/openrct2-ui/windows/Viewport.cpp:106:        auto* mainWindow = window_get_main();
src/openrct2-ui/windows/Viewport.cpp:142:                auto* mainWindow = window_get_main();
src/openrct2/ReplayManager.cpp:862:                    auto* mainWindow = window_get_main();
src/openrct2/drawing/LightFX.cpp:305:                auto* w = window_get_main();
src/openrct2/interface/Screenshot.cpp:393:        auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1800:    auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1830:    auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1860:    auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1890:    auto mainWindow = window_get_main();
src/openrct2/interface/Window_internal.cpp:20:    auto mainWindow = window_get_main();
```

Comment (Gymnasiast) [reply src/openrct2/Game.cpp:534]: I think there is a general movement towards always using `auto*` if the variable in question is a pointer in our codebase. (Would be lovely if we could enforce that automatically using clang-format or something like that.)

File: src/openrct2/ride/TrackDesign.cpp (hunk 4 above)
Comment (Broxzier) [reply src/openrct2/ride/TrackDesign.cpp:250]: `z` gets assigned again before that on line 263. The value here is completely discarded.

File: src/openrct2/Game.cpp (hunk 3 above)
Comment (Broxzier) [reply src/openrct2/Game.cpp:534]: Sounds like a good idea, that also helps finding instances that could be moved over to std::shared_ptr more easily.

File: src/openrct2/Game.cpp [hunk 5]
```diff
@@ -530,11 +528,10 @@ void game_load_init()
     {
         viewport_init_all();
         game_create_windows();
-        mainWindow = window_get_main();
     }
     else
     {
-        mainWindow = window_get_main();
+        auto* mainWindow = window_get_main();
```
Comment (Gymnasiast) [inline src/openrct2/Game.cpp:534]:
```suggestion
        const auto* mainWindow = window_get_main();
```
Or just directly feed it into `window_unfollow_sprite()`.

**Review by Gymnasiast — APPROVED** [review]

**Review by ZehMatt — APPROVED** [review]
🎡

**Pull request merged**